projects/observability/src/pages/explorer/explorer-dashboard-builder.ts

```diff
@@ -583,6 +593,7 @@ export class ExplorerDashboardBuilder {
           type: 'attribute-specification',
           attribute: attribute.name
         },
+        visible: selectedProperties.find(selectedProperty => selectedProperty.name === attribute.name) ? true : false,
```

Comment:
visible defaults to true, how was this working in the past?

If a user selects to display something, does that get preserved?

Reply:
Yes, I could ignore the field in the case that 'true' because is true by default. Answering the second question, they are shown after selecting the each metric.

Reply:
> Yes, I could ignore the field in the case that 'true' because is true by default. 

Ah - in the past this code was only being used for the added column which we wanted to display. The default ones were the ones that had hidden values. Nothing to do here - explicit as you have here is better, I was just confused on the mapping since the purpose of this code changed even though the diff shows it existed.

> Answering the second question, they are shown after selecting each metric.

Still a bit confused here - every time the query changes, wouldn't we be rebuilding the columns and tossing the user's selections (the columns they've checked off in the edit cols modal)? Or are they being preserved somewhere?